src/hooks/Circles/utils/optimisticallyUpdatePosts.ts

```diff
@@ -4,65 +4,62 @@ import type {
   PostDetailsPostQueryResponse,
 } from '../useInfinitePosts';
 import { Post } from '../types';
+import forEach from 'lodash/forEach';
+import { cloneDeep } from 'lodash';
 
 type UpdatePostsConfig = {
   queryClient: QueryClient;
-  postId: string;
-  updatePost: (post: Post) => Post;
+  id: string;
+  transformFn: (post: Post) => Post;
 };
 
 export const optimisticallyUpdatePosts = ({
   queryClient,
-  postId,
-  updatePost,
+  id,
+  transformFn,
 }: UpdatePostsConfig) => {
   queryClient.setQueryData<InfinitePostsData>(['posts'], (currentData) => {
-    const edge = findEdgeById(currentData!, postId);
-
-    if (!edge.node) {
-      return currentData!;
-    }
-
-    edge.node = updatePost(edge.node);
-
-    return currentData!;
+    const newData = cloneDeep(currentData);
+    forEach(newData?.pages, (page) => {
+      transformEdgesById(page.postsV2?.edges, id, transformFn);
+    });
+    return newData!;
   });
 
-  const visited: Post[] = [];
   queryClient
     .getQueryCache()
     .findAll({ queryKey: ['postDetails'] })
     .map((query) => {
-      query?.setData((currentData: PostDetailsPostQueryResponse) => {
-        if (!currentData) {
-          return currentData!;
-        }
-
-        const posts = extractPosts({ node: currentData.post });
-
-        const node = posts.find((post) => post.node?.id === postId);
-
-        if (node?.node && visited.indexOf(node.node) === -1) {
-          visited.push(node.node);
-          node.node = updatePost(node.node);
-        }
-
-        return currentData;
-      });
+      query?.setData(
+        (currentData: PostDetailsPostQueryResponse | undefined) => {
+          const newData = cloneDeep(currentData);
```

Comment:
Note: It appears that mutating currentData directly can result in undesired behavior. In one instance I was seeing a reaction incrementing twice. I'd recommend we always copy the data before modifying it.

Reply:
I was kind of wondering about that. Glad you caught it early. From the [docs](https://tanstack.com/query/v4/docs/react/guides/updates-from-mutation-responses#immutability):

> Updates via setQueryData must be performed in an immutable way. DO NOT attempt to write directly to the cache by mutating data (that you retrieved via from the cache) in place. It might work at first but can lead to subtle bugs along the way.

We may want to double check other areas where we do this to make sure we are covered.

Reply:
Found one more place in createPost and updated it accordingly.